cosmoz-grouped-list.js

```diff
@@ -99,20 +100,93 @@
 		 */
 		_renderedItems: null,
 
+		/**
+		 * Instances of Template that are NOT available for reuse at the moment.
+		 */
+		_usedInstances: [],
+
+		/**
+		 *  Instances of Template that are available for reuse at the moment.
+		 */
+		_reusableInstances: [],
+
+		/**
+		 * Polymer `created` livecycle function.
+		 *
+		 * @return {void}
+		 */
 		created() {
 			this._templateSelectors = [];
 			this._renderedItems = [];
-			this._physicalItens = [];
 		},
 
+		/**
+		 * Polymer `attached` livecycle function.
+		 *
+		 * @return {void}
+		 */
 		attached() {
-			this._isAttached = true;
+			this._templatesObserver = Polymer.dom(this.$.templates)
+				.observeNodes(this._onTemplatesChange.bind(this));
 			this._debounceRender();
 		},
 
+		/**
+		 * Polymer `detached` livecycle function.
+		 *
+		 * @return {void}
+		 */
 		detached() {
-			this._isAttached = false;
+			if (this._templatesObserver) {
+				Polymer.dom(this).unobserveNodes(this._templatesObserver);
+				this._templatesObserver = null;
+			}
 			this.cancelDebouncer('render');
+
+
+			this._resetAllTemplates();
+
+			this._usedInstances = [];
+			this._reusableInstances = [];
+
+		},
+		_onTemplatesChange(change) {
+			if (!Array.isArray(this._ctors) || this._ctors.length === 0) {
+				const templates = Array.from(change.addedNodes)
+					.filter(n => n.nodeType === Node.ELEMENT_NODE && n.tagName === 'TEMPLATE' && n.matches('[class]'));
```

Comment:
Could be useful to test against the expected classes `item` and `group`.
For example:
```
 n.matches('.item,.group'))
```